### ISSUE TITLE
mac os run test NetTest/3.TestBottomNeedBackward aborted

### ISSUE THREAD

**Issue by christopher5106** [BVLC/caffe#2825]
Everything compiles great, but I get the following error during 

make runtest

Where can I look to solve this issue ?
Thanks a lot for your help 

$ make runtest
.build_release/tools/caffe
caffe: command line brew
usage: caffe <command> <args>

commands:
  train           train or finetune a model
  test            score a model
  device_query    show GPU diagnostic information
  time            benchmark model execution time

  Flags from tools/caffe.cpp:
    -gpu (Run in GPU mode on given device ID.) type: int32 default: -1
    -iterations (The number of iterations to run.) type: int32 default: 50
    -model (The model definition protocol buffer text file..) type: string
      default: ""
    -snapshot (Optional; the snapshot solver state to resume training.)
      type: string default: ""
    -solver (The solver definition protocol buffer text file.) type: string
      default: ""
    -weights (Optional; the pretrained weights to initialize finetuning. Cannot
      be set simultaneously with snapshot.) type: string default: ""
.build_release/test/test_all.testbin 0 --gtest_shuffle 
Cuda number of devices: 1
Setting to use device 0
Current device id: 0
Note: Randomizing tests' orders with a seed of 64942 .
[==========] Running 1404 tests from 222 test cases.
[----------] Global test environment set-up.
[----------] 3 tests from TanHLayerTest/0, where TypeParam = N5caffe9CPUDeviceIfEE
[ RUN      ] TanHLayerTest/0.TestTanHGradient
[       OK ] TanHLayerTest/0.TestTanHGradient (491 ms)
[ RUN      ] TanHLayerTest/0.TestTanH
[       OK ] TanHLayerTest/0.TestTanH (0 ms)
[ RUN      ] TanHLayerTest/0.TestTanHOverflow
[       OK ] TanHLayerTest/0.TestTanHOverflow (0 ms)
[----------] 3 tests from TanHLayerTest/0 (491 ms total)

[----------] 1 test from GaussianFillerTest/1, where TypeParam = d
[ RUN      ] GaussianFillerTest/1.TestFill
[       OK ] GaussianFillerTest/1.TestFill (1 ms)
[----------] 1 test from GaussianFillerTest/1 (1 ms total)

[----------] 3 tests from BlobMathTest/0, where TypeParam = N5caffe9CPUDeviceIfEE
[ RUN      ] BlobMathTest/0.TestScaleData
[       OK ] BlobMathTest/0.TestScaleData (0 ms)
[ RUN      ] BlobMathTest/0.TestAsum
[       OK ] BlobMathTest/0.TestAsum (0 ms)
[ RUN      ] BlobMathTest/0.TestSumOfSquares
[       OK ] BlobMathTest/0.TestSumOfSquares (0 ms)
[----------] 3 tests from BlobMathTest/0 (0 ms total)

[----------] 3 tests from SplitLayerTest/2, where TypeParam = N5caffe9GPUDeviceIfEE
[ RUN      ] SplitLayerTest/2.Test
[       OK ] SplitLayerTest/2.Test (0 ms)
[ RUN      ] SplitLayerTest/2.TestSetup
[       OK ] SplitLayerTest/2.TestSetup (0 ms)
[ RUN      ] SplitLayerTest/2.TestGradient
[       OK ] SplitLayerTest/2.TestGradient (48 ms)
[----------] 3 tests from SplitLayerTest/2 (48 ms total)

[----------] 6 tests from ArgMaxLayerTest/1, where TypeParam = d
[ RUN      ] ArgMaxLayerTest/1.TestSetup
[       OK ] ArgMaxLayerTest/1.TestSetup (0 ms)
[ RUN      ] ArgMaxLayerTest/1.TestCPU
[       OK ] ArgMaxLayerTest/1.TestCPU (0 ms)
[ RUN      ] ArgMaxLayerTest/1.TestCPUMaxVal
[       OK ] ArgMaxLayerTest/1.TestCPUMaxVal (0 ms)
[ RUN      ] ArgMaxLayerTest/1.TestCPUTopK
[       OK ] ArgMaxLayerTest/1.TestCPUTopK (0 ms)
[ RUN      ] ArgMaxLayerTest/1.TestCPUMaxValTopK
[       OK ] ArgMaxLayerTest/1.TestCPUMaxValTopK (1 ms)
[ RUN      ] ArgMaxLayerTest/1.TestSetupMaxVal
[       OK ] ArgMaxLayerTest/1.TestSetupMaxVal (0 ms)
[----------] 6 tests from ArgMaxLayerTest/1 (1 ms total)

[----------] 6 tests from CuDNNConvolutionLayerTest/1, where TypeParam = d
[ RUN      ] CuDNNConvolutionLayerTest/1.TestSimpleConvolutionCuDNN
[       OK ] CuDNNConvolutionLayerTest/1.TestSimpleConvolutionCuDNN (129 ms)
[ RUN      ] CuDNNConvolutionLayerTest/1.TestSobelConvolutionCuDNN
[       OK ] CuDNNConvolutionLayerTest/1.TestSobelConvolutionCuDNN (3 ms)
[ RUN      ] CuDNNConvolutionLayerTest/1.TestSimpleConvolutionGroupCuDNN
[       OK ] CuDNNConvolutionLayerTest/1.TestSimpleConvolutionGroupCuDNN (1 ms)
[ RUN      ] CuDNNConvolutionLayerTest/1.TestGradientGroupCuDNN
[       OK ] CuDNNConvolutionLayerTest/1.TestGradientGroupCuDNN (520 ms)
[ RUN      ] CuDNNConvolutionLayerTest/1.TestSetupCuDNN
[       OK ] CuDNNConvolutionLayerTest/1.TestSetupCuDNN (1 ms)
[ RUN      ] CuDNNConvolutionLayerTest/1.TestGradientCuDNN
[       OK ] CuDNNConvolutionLayerTest/1.TestGradientCuDNN (1183 ms)
[----------] 6 tests from CuDNNConvolutionLayerTest/1 (1837 ms total)

[----------] 12 tests from ReshapeLayerTest/2, where TypeParam = N5caffe9GPUDeviceIfEE
[ RUN      ] ReshapeLayerTest/2.TestFlattenMiddle
[       OK ] ReshapeLayerTest/2.TestFlattenMiddle (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestFlattenValues
[       OK ] ReshapeLayerTest/2.TestFlattenValues (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestForwardAfterReshape
[       OK ] ReshapeLayerTest/2.TestForwardAfterReshape (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestInsertSingletonAxesStart
[       OK ] ReshapeLayerTest/2.TestInsertSingletonAxesStart (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestCopyDimensions
[       OK ] ReshapeLayerTest/2.TestCopyDimensions (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestInsertSingletonAxesMiddle
[       OK ] ReshapeLayerTest/2.TestInsertSingletonAxesMiddle (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestGradient
[       OK ] ReshapeLayerTest/2.TestGradient (2 ms)
[ RUN      ] ReshapeLayerTest/2.TestInsertSingletonAxesEnd
[       OK ] ReshapeLayerTest/2.TestInsertSingletonAxesEnd (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestFlattenOutputSizes
[       OK ] ReshapeLayerTest/2.TestFlattenOutputSizes (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestForward
[       OK ] ReshapeLayerTest/2.TestForward (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestInferenceOfUnspecifiedWithStartAxis
[       OK ] ReshapeLayerTest/2.TestInferenceOfUnspecifiedWithStartAxis (0 ms)
[ RUN      ] ReshapeLayerTest/2.TestInferenceOfUnspecified
[       OK ] ReshapeLayerTest/2.TestInferenceOfUnspecified (0 ms)
[----------] 12 tests from ReshapeLayerTest/2 (2 ms total)

[----------] 10 tests from EltwiseLayerTest/1, where TypeParam = N5caffe9CPUDeviceIdEE
[ RUN      ] EltwiseLayerTest/1.TestProd
[       OK ] EltwiseLayerTest/1.TestProd (1 ms)
[ RUN      ] EltwiseLayerTest/1.TestSumCoeff
[       OK ] EltwiseLayerTest/1.TestSumCoeff (0 ms)
[ RUN      ] EltwiseLayerTest/1.TestUnstableProdGradient
[       OK ] EltwiseLayerTest/1.TestUnstableProdGradient (3 ms)
[ RUN      ] EltwiseLayerTest/1.TestSumCoeffGradient
[       OK ] EltwiseLayerTest/1.TestSumCoeffGradient (4 ms)
[ RUN      ] EltwiseLayerTest/1.TestMax
[       OK ] EltwiseLayerTest/1.TestMax (0 ms)
[ RUN      ] EltwiseLayerTest/1.TestSumGradient
[       OK ] EltwiseLayerTest/1.TestSumGradient (3 ms)
[ RUN      ] EltwiseLayerTest/1.TestMaxGradient
[       OK ] EltwiseLayerTest/1.TestMaxGradient (4 ms)
[ RUN      ] EltwiseLayerTest/1.TestSetUp
[       OK ] EltwiseLayerTest/1.TestSetUp (0 ms)
[ RUN      ] EltwiseLayerTest/1.TestSum
[       OK ] EltwiseLayerTest/1.TestSum (0 ms)
[ RUN      ] EltwiseLayerTest/1.TestStableProdGradient
[       OK ] EltwiseLayerTest/1.TestStableProdGradient (3 ms)
[----------] 10 tests from EltwiseLayerTest/1 (18 ms total)

[----------] 8 tests from ConvolutionLayerTest/3, where TypeParam = N5caffe9GPUDeviceIdEE
[ RUN      ] ConvolutionLayerTest/3.TestSimpleConvolutionGroup
[       OK ] ConvolutionLayerTest/3.TestSimpleConvolutionGroup (1 ms)
[ RUN      ] ConvolutionLayerTest/3.Test1x1Gradient
[       OK ] ConvolutionLayerTest/3.Test1x1Gradient (9360 ms)
[ RUN      ] ConvolutionLayerTest/3.TestSobelConvolution
[       OK ] ConvolutionLayerTest/3.TestSobelConvolution (1 ms)
[ RUN      ] ConvolutionLayerTest/3.TestGradient
[       OK ] ConvolutionLayerTest/3.TestGradient (1613 ms)
[ RUN      ] ConvolutionLayerTest/3.TestSimpleConvolution
[       OK ] ConvolutionLayerTest/3.TestSimpleConvolution (0 ms)
[ RUN      ] ConvolutionLayerTest/3.TestGradientGroup
[       OK ] ConvolutionLayerTest/3.TestGradientGroup (479 ms)
[ RUN      ] ConvolutionLayerTest/3.TestSetup
[       OK ] ConvolutionLayerTest/3.TestSetup (0 ms)
[ RUN      ] ConvolutionLayerTest/3.Test1x1Convolution
[       OK ] ConvolutionLayerTest/3.Test1x1Convolution (1 ms)
[----------] 8 tests from ConvolutionLayerTest/3 (11455 ms total)

[----------] 22 tests from NetTest/3, where TypeParam = N5caffe9GPUDeviceIdEE
[ RUN      ] NetTest/3.TestBottomNeedBackward
*** Aborted at 1438003120 (unix time) try "date -d @1438003120" if you are using GNU date ***
PC: @     0x7fff83579820 std::__1::__shared_count::__add_shared()
*** SIGSEGV (@0x8) received by PID 72901 (TID 0x7fff74b19300) stack trace: ***
    @     0x7fff87a6cf1a _sigtramp
    @     0x7fff67419764 (unknown)
    @     0x7fff83546bc0 std::__1::ios_base::getloc()
    @        0x107c03851 caffe::Net<>::Init()
    @        0x107c03217 caffe::Net<>::Net()
    @        0x101643064 caffe::NetTest<>::InitNetFromProtoString()
    @        0x101643653 caffe::NetTest<>::InitTinyNet()
    @        0x10169edfd caffe::NetTest_TestBottomNeedBackward_Test<>::TestBody()
    @        0x101878e0c testing::internal::HandleExceptionsInMethodIfSupported<>()
    @        0x10186756a testing::Test::Run()
    @        0x101868492 testing::TestInfo::Run()
    @        0x101868b80 testing::TestCase::Run()
    @        0x10186f336 testing::internal::UnitTestImpl::RunAllTests()
    @        0x101879614 testing::internal::HandleExceptionsInMethodIfSupported<>()
    @        0x10186f049 testing::UnitTest::Run()
    @        0x101426f6d main
    @     0x7fff839a15c9 start
make: *** [runtest] Segmentation fault: 11

**Comment by christopher5106** [comment]
Another one : 
[----------] 7 tests from NesterovSolverTest/2, where TypeParam = N5caffe9GPUDeviceIfEE
[ RUN      ] NesterovSolverTest/2.TestNesterovLeastSquaresUpdate
*** Aborted at 1438003553 (unix time) try "date -d @1438003553" if you are using GNU date ***
PC: @     0x7fff83579820 std::__1::__shared_count::__add_shared()
*** SIGSEGV (@0x8) received by PID 73163 (TID 0x7fff74b19300) stack trace: ***
    @     0x7fff87a6cf1a _sigtramp
    @        0x11086a726 fatbinData
    @     0x7fff83546bc0 std::__1::ios_base::getloc()
    @        0x110564280 caffe::Solver<>::Init()
    @        0x11056419d caffe::Solver<>::Solver()
    @        0x109ec03e2 caffe::SGDSolver<>::SGDSolver()
    @        0x109ec007b caffe::NesterovSolverTest<>::InitSolver()
    @        0x109ec0369 caffe::GradientBasedSolverTest<>::InitSolverFromProtoString()
    @        0x109ec184d caffe::GradientBasedSolverTest<>::RunLeastSquaresSolver()
    @        0x109ec8e49 caffe::GradientBasedSolverTest<>::TestLeastSquaresUpdate()
    @        0x10a1d6e0c testing::internal::HandleExceptionsInMethodIfSupported<>()
    @        0x10a1c556a testing::Test::Run()
    @        0x10a1c6492 testing::TestInfo::Run()
    @        0x10a1c6b80 testing::TestCase::Run()
    @        0x10a1cd336 testing::internal::UnitTestImpl::RunAllTests()
    @        0x10a1d7614 testing::internal::HandleExceptionsInMethodIfSupported<>()
    @        0x10a1cd049 testing::UnitTest::Run()
    @        0x109d84f6d main
    @     0x7fff839a15c9 start
make: *** [runtest] Segmentation fault: 11

**Comment by christopher5106** [comment]
Everything works for me now 

In any case, here is my full install : 
http://christopher5106.github.io/big/data/2015/07/16/deep-learning-install-caffe-cudnn-cuda-for-digits-python-on-mac-osx.html
